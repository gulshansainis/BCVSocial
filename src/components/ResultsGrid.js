import React from "react";

const ResultsGrid = ({ results }) => {
  function formatDate(date) {
    const year = date.getFullYear();

    let month = (1 + date.getMonth()).toString();
    month = month.length > 1 ? month : `0${month}`;

    let day = date.getDate().toString();
    day = day.length > 1 ? day : `0${day}`;

    return month + "/" + day + "/" + year;
  }

  return (
    <ul className="resultgrid">
      <li>
        <span>
          <strong>Date</strong>
        </span>
        <span>
          <strong>Venue</strong>
        </span>
        <span>
          <strong>Team A</strong>
        </span>
        <span>
          <strong>Team B</strong>
        </span>
        <span>
          <strong>Winner</strong>
        </span>
      </li>
      {results.map(result => {
        const {
          venue,
          fifa_id,
          home_team_country: teamA,
          away_team_country: teamB,
          winner,
          datetime
        } = result;

        return (
          <ResultRow
            key={fifa_id}
            date={formatDate(new Date(datetime))}
            venue={venue}
            teamA={teamA}
            teamB={teamB}
            winner={winner === "Draw" ? "Match draw" : `${winner} won`}
          />
        );
      })}
    </ul>
  );
};

const ResultRow = ({ date, venue, teamA, teamB, winner }) => {
  return (
    <li>
      <span>{date}</span>
      <span>{venue}</span>
      <span>{teamA}</span>
      <span>{teamB}</span>
      <span>{winner}</span>
    </li>
  );
};

export default ResultsGrid;
