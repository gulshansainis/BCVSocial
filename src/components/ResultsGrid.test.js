import React from "react";
import renderer from "react-test-renderer";
import ResultsGrid from "./ResultsGrid";

it("renders correctly", () => {
  const data = [
    {
      venue: "Valenciennes",
      fifa_id: "300438227",
      home_team_country: "Australia",
      away_team_country: "Italy",
      datetime: "2019-06-09T23:00:00Z",
      winner: "Italy",
      winner_code: "ITA"
    }
  ];
  const tree = renderer.create(<ResultsGrid results={data} />).toJSON();
  expect(tree).toMatchSnapshot();
});
