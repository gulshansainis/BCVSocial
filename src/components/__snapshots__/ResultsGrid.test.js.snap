// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h2>
    Match Results
  </h2>,
  <label
    htmlFor="colorPicker"
  >
    Configure results background color
     
    <input
      onChange={[Function]}
      type="color"
    />
  </label>,
  <ul
    className="resultgrid"
    style={
      Object {
        "background": "",
      }
    }
  >
    <li>
      <span>
        <strong>
          Date
        </strong>
      </span>
      <span>
        <strong>
          Venue
        </strong>
      </span>
      <span>
        <strong>
          Team A
        </strong>
      </span>
      <span>
        <strong>
          Team B
        </strong>
      </span>
      <span>
        <strong>
          Winner
        </strong>
      </span>
    </li>
    <li>
      <span>
        06/10/2019
      </span>
      <span>
        Valenciennes
      </span>
      <span>
        Australia
      </span>
      <span>
        Italy
      </span>
      <span>
        Italy won
      </span>
    </li>
  </ul>,
]
`;
